x-pack/solutions/observability/plugins/observability/server/services/referenced_panel_manager.ts

```diff
@@ -14,23 +14,51 @@ export class ReferencedPanelManager {
   // So, if the same saved object panel is referenced in two different dashboards, it will have different panelIndex values in each dashboard, but the same panelId, since they're both referencing the same panel.
   private panelsById = new Map<string, ReferencedPanelAttributesWithReferences>();
   private panelIndexToId = new Map<string, string>();
+  private panelsTypeById = new Map<string, string>();
```

Comment:
This looks great, but I'm just confused a bit again by the variable names. Maybe `panelsToFetchById`. Maybe not the best. I just don't understand what 'type' means here semantically.

Reply:
The key of the map stores the panel ID and the value stores the panel type, I'll need both to fetch the panels.
That's why I called this map `panelsTypeById`, it follows the `valueBykey` pattern and reflects what’s stored. 

What is confusing you?